subprojects/stdlib/library_api.h

```diff
@@ -0,0 +1,14 @@
+#ifndef __LIBRARY_API_H__
+#define __LIBRARY_API_H__
+
+#include <extern_library_helper.h>
+
+#define FUNCTION(name, func) (struct extern_function){name, (function_pointer) func}
+
+#define DEFINE_LIBRARY(...)                                              \
+    struct extern_function* expose_library() {                           \
+        static struct extern_function lib[] = {__VA_ARGS__, {0, 0}};     \
```

Comment:
To keep a bit more future proof

```suggestion
        static struct extern_function lib[] = {__VA_ARGS__, { }};     \
```